core/common/src/geometry/ElementGeometry.ts

```diff
@@ -791,21 +794,26 @@ export namespace ElementGeometry {
 
     let data;
     const entityData = ppfb.entityDataArray();
-    if (wantBRepData && null !== entityData)
-      data = Base64.fromUint8Array(entityData);
+    if (wantBRepData && null !== entityData) {
+      const base64Header = "encoding=base64;";
+      data = base64Header + Base64.fromUint8Array(entityData);
+    }
 
     return { data, type, transform: transform?.toJSON(), faceSymbology };
   }
 
-  /** Create entry from a [[BRepEntity.DataProps]] */
+  /** Create entry from a [[BRepEntity.DataProps]]. Provided for compatibility with GeometryStreamBuilder only. */
   export function fromBRep(brep: BRepEntity.DataProps): ElementGeometryDataEntry | undefined {
     const fbb = new flatbuffers.Builder();
     const builder = EGFBAccessors.BRepData;
     let dataOffset;
     let faceSymbOffset;
 
     if (undefined !== brep.data) {
-      const entityData = Base64.toUint8Array(brep.data);
+      const base64Header = "encoding=base64;";
+      if (brep.data.length < base64Header.length || !brep.data.startsWith(base64Header))

```

Comment:
will this test work before a new backend is built?

Reply:
It's not currently being called, and toUint8Array wasn't pleased with the bad geometry stream from getElementProps already. The new test that was written to excercise this code (and discovered the problem) currently has the call to fromBRep commented off until the addon is updated.